OracleWebLogic/samples/12212-msiserver/README.md

```diff
@@ -0,0 +1,160 @@
+run a managed with a name managedServer<RandomNumber>
+
+docker build -t 12212-msiserver --build-arg ms_name_prefix=managedServer .
+
+**Build Argument number_of_ms**
+By default, this image comes configured with 10 managed servers, ms1 to ms10. However, the image
+can be built with configurable number of managed servers using NUMBER_OF_MS argument
+
+docker build -t 12212-msiserver --build-arg number_of_ms=15 .
+
+**Other build arguments**
+* domain_name may be used to identify the name of generated domain that gets packed into MSI image. Default value msi-sample
+* domains_dir may be used identify the directory under $ORACLE_HOME where the domain home directory is created. Default value wlserver/samples/domains 
+* ms_port may be used to configure port of the managed server, default is 8011
+* prod_or_dev may be used to identify whether server is started in production or development mode. Defaults to "dev" for development mode.
+
+**Runtime Argument MS_NAME**
+This argument may be used to completely override the managed server name.
+For example, the following command may be used to run a managed server with name ms1
+
+docker run --name msiserver --env MS_NAME=managedServer1 12212-msiserver
+
+How to use the base image to add application
+--------------------------------------------
+To build, run:
+        $ docker build -t 12212-summercamps-msiserver -f Dockerfile.addapp --build-arg name=summercamps  --build-arg source=apps/summercamps.ear --build-arg simple_filename=summercamps.ear .
+
+Dockerfile.addapp adds an app called summercamps.ear targeted to the cluster that
+the managed server is a part of
+
+To start the Managed Server in MSI mode with this app, run:
+
+        $ docker run -d -p 8011:8011 12212-summercamps-msiserver
+
+summercamps app will now be accessible at
+
+http://localhost:8011/
+
+As with the base image, you can still override managed server name and cluster name
+**Build arguments**
+Three build arguments may be used to customize the image to include an application of user
+choice. By default, the build arguments point to an example application included in this
+sample. The "name" argument helps identify the name of deployment, while the "source"
+argument helps identify the source of the application. The source is copied into the image.
+So "simple_filename" helps identify the name of the file where the source is copied to
+
+Using swarm service creation with this image
+--------------------------------------------
+The image called 12212-summercamps-msiserver can be used for Docker service creation
+to scale out to multiple replicas. docker service creation can only be done
+using an image from a docker registry. If you don't have publishing rights to a docker
+registry you can start one locally. More details can be found here
+https://docs.docker.com/registry/deploying/
+
+Here are the three steps you'll need to start, use and push to your local registry.

```

Comment:
Don't provide documentation on how to run a registry, because this will get out of date. You already point to the Docker docs, that's sufficient. 

Also, a registry is not required for a service, if you have the same image on all the hosts in the Swarm. It's just that the simplest way for that to happen is to use a local registry. :)